bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/BigtableConnection.java

```diff
@@ -263,9 +269,46 @@ public RegionLocator getRegionLocator(TableName tableName) throws IOException {
     throw new IllegalStateException(newLocator + " was supposed to be in the cache");
   }
 
+  static final MethodHandler ADMIN_HANDLER = new MethodHandler() {
+    @Override
+    public Object invoke(Object self, Method thisMethod, Method proceed, Object[] args)
+        throws Throwable {
+      throw new UnsupportedOperationException(thisMethod.getName() + " not supported");
+    }
+  };
+
+  static final ProxyFactory ADMIN_FACTORY = new ProxyFactory();
+  static {
+    ADMIN_FACTORY.setSuperclass(BigtableAdmin.class);
+    ADMIN_FACTORY.setFilter(
+        new MethodFilter() {
+            @Override
+            public boolean isHandled(Method method) {
+                return Modifier.isAbstract(method.getModifiers());
+            }
+        }
+    );
+  }
+
+  static final Class<?>[] ADMIN_CONSTRUCTOR_TYPES = {
+      BigtableOptions.class,
+      Configuration.class,
+      BigtableConnection.class,
+      BigtableTableAdminClient.class,
+      Set.class
+  };
+
   @Override
+  // Admin has different methods depending on which hbase version contains it.  Bigtable doesn't
+  // support a lot of methods in hbase Admin, since they do not apply to Bigtable's operations.
+  // for those methods, throw UnsupportedException.
   public Admin getAdmin() throws IOException {
-    return new BigtableAdmin(options, conf, this, bigtableTableAdminClient, this.disabledTables);
+    Object[] methodParams = { options, conf, this, bigtableTableAdminClient, this.disabledTables };
+    try {
+      return (Admin) ADMIN_FACTORY.create(ADMIN_CONSTRUCTOR_TYPES, methodParams, ADMIN_HANDLER);
```

Comment:
This seems to work. Can you create test cases to test calls are forwarded to the super class and rejected when method is not implemented?